telethon/network/mtprotosender.py

```diff
@@ -331,6 +332,7 @@ async def _disconnect(self, error=None):
         if self._disconnected and not self._disconnected.done():
             if error:
                 self._disconnected.set_exception(error)
+                self._disconnected.exception()
```

Comment:
Again this is silencing the exception.

Reply:
@Lonami The problem is that exceptions that come here from e.g. `ConnectionError` inside `MTProtoSender` cannot be handled by user, since future `self._disconnected` of `MTProtoSender`has no public interface. So you cannot really recover this exception outside of the `MTProtoSender` class.

Solution would be either to silence it or to let user have some public interface to this future to extract exception.